src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs

```diff
@@ -274,6 +276,20 @@ static virtual TSelf CreateTruncating<TOther>(TOther value)
         /// <exception cref="OverflowException"><paramref name="s" /> is not representable by <typeparamref name="TSelf" />.</exception>
         static abstract TSelf Parse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider);
 
+        /// <summary>Parses a span of UTF-8 characters into a value.</summary>
+        /// <param name="utf8Text">The span of UTF-8 characters to parse.</param>
+        /// <param name="style">A bitwise combination of number styles that can be present in <paramref name="utf8Text" />.</param>
+        /// <param name="provider">An object that provides culture-specific formatting information about <paramref name="utf8Text" />.</param>
+        /// <returns>The result of parsing <paramref name="utf8Text" />.</returns>
+        /// <exception cref="ArgumentException"><paramref name="style" /> is not a supported <see cref="NumberStyles" /> value.</exception>
+        /// <exception cref="FormatException"><paramref name="utf8Text" /> is not in the correct format.</exception>
+        /// <exception cref="OverflowException"><paramref name="utf8Text" /> is not representable by <typeparamref name="TSelf" />.</exception>
+        static virtual TSelf Parse(ReadOnlySpan<byte> utf8Text, NumberStyles style, IFormatProvider? provider)
+        {
+            string s = Encoding.Unicode.GetString(utf8Text);
```

Comment:
Same allocation question.  I realize this won't affect any of our own primitives, since we'll override this on all of them, but it'd be nice if for anyone else implementing these if this didn't allocate a full copy of the input.  Could we stackalloc for shorter inputs at least?  Presumably that would handle most numerical inputs being parsed.

Reply:
Fixed to use stackalloc for less than or equal to 256 characters and ArrayPool otherwise.